File: notefronte/src/app-students-interfaces/app/models/EtudiantNoteMaxDto.ts
export class EtudiantNoteMaxDto {
  constructor(
    public ueCode: string = '',
    public ueLibelle: string = '',
    public devoir: number = 0,
    public examen: number = 0,
    public typeUe: string = '',
    public semestre: string = '',
  ) {}
}

File: Back/src/main/java/com/defitech/GestUni/dto/Note/EtudiantNoteMaxDto.java
package com.defitech.GestUni.dto.Note;
import com.defitech.GestUni.enums.TypeNote;
import com.defitech.GestUni.enums.TypeSemestre;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class EtudiantNoteMaxDto {
    private String ueCode;
    private String ueLibelle;
    private float devoir;
    private float examen;
    private String typeUe;
    private TypeSemestre semestre;

}
